.github/workflows/push.yml

```diff
@@ -1,35 +1,47 @@
 name: Docker build and push
-on:   
+on:
   push:
     branches:
       - master
+  pull_request:
   schedule:
     - cron: "0 2 * * 0"
 
 env:
-  IMAGE_NAME: nodejs
+  DOCKER_BUILDKIT: 1
 
 jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: hadolint/hadolint-action@v2.0.0
+        with:
+          dockerfile: Dockerfile
+          config: hadolint-ci.yaml
+
   push:
     runs-on: ubuntu-latest
+    needs: lint
+    if: github.event_name != 'pull_request'
     permissions:
       packages: write
       contents: read
 
     strategy:
       matrix:
-        node: [10, 12, 14]
+        node: [12, 14, 16]
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
 
-      - uses: hadolint/hadolint-action@v1.5.0
-        with:
-          dockerfile: Dockerfile
-          config: hadolint-ci.yaml
+      - name: Set image name
+        run: |
+          echo "DOCKER_IMAGE=ghcr.io/${GITHUB_REPOSITORY_OWNER,,}/nodejs" >> ${GITHUB_ENV}
 
       - name: Build image
-        run: docker-compose -f docker-compose.yml build node${{ matrix.node }}
+        run: IMAGE=${DOCKER_IMAGE} docker-compose -f docker-compose.yml build node${{ matrix.node }}
```

Comment:
Why not simply hardcode the image name in docker-compose.yml and then do two tags based on that using the env var?